### PR TITLE
[feat] - dojot acl service

### PR DESCRIPTION
**Pull request by jonaphael**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Is there any issue related to this PR? (Link to an issue, e.g. #99999)** 

* **Other information**:

### CODE REVIEW

File: connector/dojot_acl/.eslintignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Ignores the Nearley compiled parser
+app/WhereProcessing/Parser.js
```
Comment (mprevide) [inline connector/dojot_acl/.eslintignore:2]: Remove `app/WhereProcessing/Parser.js`

File: .github/workflows/dojot_acl.yml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      uses: actions/checkout@v1
+
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+
+    - name: npm install, build, lint and test
+      run: |
+        cd connector/dojot_acl
```
Comment (mprevide) [inline .github/workflows/dojot_acl.yml:34]: Maybe It should be inside of verne folder?

File: connector/dojot_acl/config/default.conf [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+# KAFKA CONSUMER
+consumer.group.id=dojot-acl
+consumer.metadata.broker.list=kafka:9092
+
+# TOPIC
+topic.auto.offset.reset=largest
+
+#
+# https://github.com/dojot/dojot/blob/b4a6328164610c1d5213bec31e82ed84a80b27bf/iam/x509-identity-mgmt/js/config/default.conf#L404
```
Comment (mprevide) [inline connector/dojot_acl/config/default.conf:28]: Remove these URLs

File: connector/dojot_acl/app/kafka/KafkaConsumer.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+  /**
+     * Unregister all callbacks
+     */
+  unregisterAllCallbacks() {
+    logger.debug('unregisterCallbacks: Unregister All Callbacks');
+
+    this.registeredCallbacks.forEach((callbackId) => this.consumer.unregisterCallback(callbackId));
+    this.registeredCallbacks.clear();
+  }
```
Comment (thiagodpf) [inline connector/dojot_acl/app/kafka/KafkaConsumer.js:82]: What is the need for these methods, it seems to me that they are not used anywhere ...

**Review by thiagodpf — COMMENTED** [review]
Some comments...

File: connector/dojot_acl/index.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+ * @file index.js
+ * @dependencies [ util, @dojot/microservice-sdk ]
+ */
+
+const util = require('util');
+const { ConfigManager, Logger } = require('@dojot/microservice-sdk');
+
+const DOJOT_ACL_CONFIG_LABEL = 'DOJOT_ACL';
+
+const userConfigFile = process.env.DOJOT_ACL_APP_USSER_CONFIG_FILE || 'production.conf';
```
Comment (mprevide) [inline connector/dojot_acl/index.js:11]: `DOJOT_ACL_APP_USER_CONFIG_FILE`?

Comment (mprevide) [reply connector/dojot_acl/index.js:11]: or just  `DOJOT_ACL_USER_CONFIG_FILE`?

File: connector/dojot_acl/index.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+ConfigManager.loadSettings(DOJOT_ACL_CONFIG_LABEL, userConfigFile);
+
+const config = ConfigManager.getConfig(DOJOT_ACL_CONFIG_LABEL);
+
+Logger.setTransport('console', { level: config.log['console.level'] });
```
Comment (mprevide) [inline connector/dojot_acl/index.js:17]: To follow the other services in this repository, you may be able to configure options for log in  files.

File: connector/dojot_acl/samples/kafkacat.producer.sample [hunk 7]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
```
Comment (mprevide) [inline connector/dojot_acl/samples/kafkacat.producer.sample:1]: I think you could create a ReadMe for the example, explaining how to run it...

File: connector/dojot_acl/app/App.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+
+    const topics = new RegExp(`^.+\\.${topicSuffix}`);
+    kafkaConsumer.registerCallback(topics, incommingMessagesCallback);
+
+    // healthchecker and shutdown
+    const healthCheckerInterval = config.healthcheck['kafka.interval.ms'];
+    StateManager.addHealthChecker('kafka', kafkaConsumer.checkHealth.bind(kafkaConsumer), healthCheckerInterval);
+    StateManager.registerShutdownHandler(kafkaConsumer.shutdownProcess.bind(kafkaConsumer));
+  }).catch((err) => {
+    logger.error(`Error initializing the kafka consumer exiting!... ${err}`);
```
Comment (mprevide) [inline connector/dojot_acl/app/App.js:85]: I'm not sure if this process exit 1 will work as expected.

File: connector/dojot_acl/app/kafka/KafkaConsumer.js (hunk 4 above)
Comment (mprevide) [reply connector/dojot_acl/app/kafka/KafkaConsumer.js:82]: could be used in graceful shutdown

Comment (mprevide) [reply connector/dojot_acl/app/kafka/KafkaConsumer.js:82]: could be used in graceful shutdown

File: connector/dojot_acl/app/App.js [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+    logger.info('Kafka Consumer initialized sucessfully!');
+
+    const topicSuffix = config.app['consumer.topic.suffix'].replace(/\./g, '\\.');
+
+    const topics = new RegExp(`^.+\\.${topicSuffix}`);
+    kafkaConsumer.registerCallback(topics, incommingMessagesCallback);
+
+    // healthchecker and shutdown
+    const healthCheckerInterval = config.healthcheck['kafka.interval.ms'];
+    StateManager.addHealthChecker('kafka', kafkaConsumer.checkHealth.bind(kafkaConsumer), healthCheckerInterval);
```
Comment (mprevide) [inline connector/dojot_acl/app/App.js:82]: Maybe you could follow a pattern ... In the case of kafka you register heath and shutdown here, and in the case of Redis you do it within the redis class.

File: .github/workflows/dojot_acl.yml [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+        name: ${{ steps.get_owner.outputs.owner }}/dojot-acl
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+        workdir: connector/dojot_acl
+        # Label with the built image's commit
+        buildoptions: " --label 'commit_sha'='${{ github.sha }}' "
+        # Whether tag will be based in the branch name or in the tag name
+        tag_names: true
+
+  gh-pages:
```
Comment (mprevide) [inline .github/workflows/dojot_acl.yml:79]: I think you can remove the gh-pages step, because there's no api documentation in this service right now.

**Comment by mprevide** [comment]
Please, associate this PR with a issue.

File: connector/dojot_acl/app/kafka/KafkaConsumer.js (hunk 4 above)
Comment (thiagodpf) [reply connector/dojot_acl/app/kafka/KafkaConsumer.js:82]: Is it necessary to call this at shutdown? 
I saw that he already calls method `this.consumer.finish()` on shutdown process, which in turn performs the unsubscription of all topics... 
This is a question of mine too, I don't know the library well.

Comment (mprevide) [reply connector/dojot_acl/app/kafka/KafkaConsumer.js:82]: I think not... 
Before the finish was not implemented.

**Review by rascaraficci — COMMENTED** [review]
I left some comments.

File: .github/workflows/dojot_acl.yml [hunk 11]
```diff
@@ -0,0 +1,113 @@
+name: Dojot ACL (connector/dojot_acl)
```
[the remark this answers is not in the document]
Comment (thiagodpf) [reply .github/workflows/dojot_acl.yml:1]: Your suggestion seems much more coherent to me!

File: .github/workflows/x509-ejbca.yml [hunk 12]
```diff
@@ -0,0 +1,34 @@
+name: x509-ejbca
```
Comment (mprevide) [inline .github/workflows/x509-ejbca.yml:1]: I think it not be part of your PR

**Pull request closed without merging**